src/composables/transactionList.ts

```diff
@@ -0,0 +1,337 @@
+        setPendingTransactionSentByHash(address, hash);
+      } catch (e) {
+        removePendingTransactionByAccount(address, hash);
+      }
+    }
+  }
+
+  function addCustomPendingTransactionForAccount(
+    address: Encoded.AccountAddress,
+    transaction: ITransaction,
+  ) {
+    createTransactionsState(address);
+    transactions.value[address].pending[nodeNetworkId.value!] = [transaction];
+    waitTransactionMined(address, transaction.hash);
+  }
+
+  async function fetchTransactionsFromMiddleware(
+    transactionState: ITransactionsState,
+    address: Encoded.AccountAddress,
+    recent: boolean,
+    limit: number,
+  ) {
+    const url = (recent || transactionState.nextPageUrl === '')
+      ? `/v2/accounts/${address}/activities?limit=${limit}`
+      : transactionState.nextPageUrl!;
+
+    try {
+      const { data, next } = await fetchFromMiddlewareCamelCased(url);
+      if (!recent) {
+        setTransactionsNextPage(address, next);
+      }
+      return data;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchPendingTransactions(address: Encoded.AccountAddress) {
+    const sdk = await getAeSdk();
+
+    try {
+      const fetchedPendingTransaction = (
+        await sdk.api.getPendingAccountTransactionsByPubkey(address)
+      );
+      const transactionState = getAccountTransactionsState(address);
+
+      return JsonBig.parse(JsonBig.stringify(
+        fetchedPendingTransaction?.transactions || [],
+      ))
+        .filter((transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]
+          ?.find((tx) => tx?.hash === transaction?.hash))
+        .map((transaction: ITransaction) => ({ ...transaction, pending: true }));
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
+    try {
+      await getAeSdk();
+      const response = await fetchJson(
+        `${nodeNetworkId.value}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
+      );
+      if (response.message) {
+        return [];
+      }
+      // TODO prepare interface for response
+      const tipWithdrawnTransactions: ITransaction[] = (response as any[]).map(({
+        amount,
+        contract,
+        height,
+        data: { tx },
+        ...t
+      }) => ({
+        tx: {
+          ...tx,
+          address,
+          amount,
+          contractId: contract,
+          type: Tag[Tag.ContractCallTx],
+        },
+        ...t,
+        microTime: new Date(t.createdAt).getTime(),
+        blockHeight: height,
+        claim: true,
+      }));
+
+      return tipWithdrawnTransactions;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchTransactions(
+    limit: number,
+    recent: boolean,
+    address: Encoded.AccountAddress,
+  ): Promise<ITransaction[]> {
+    await getAeSdk();
+    createTransactionsState(address);
```

Comment:
This can be removed as we don't need to be sure the state exists.

Reply:
if this object is not exist, a lot of things below not working properly.,like:
`    
transactions.value[address].loaded = uniqBy(
      [...transactions.value[address].loaded, ...preparedTransactions],
      'hash',
    );
`
or 

`
    const oldPendingTransactionForAccount: ITransaction[] = (
      transactions.value[address]?.pending[nodeNetworkId.value!] || []
    );
`

Reply:
Uhhh, makes sense. Resolving.